### DISCUSSION TITLE
How to log Set and Map

### DISCUSSION THREAD

**Discussion by mmakara2021** [jongpie/NebulaLogger#221] [Q&A] [answered]
Hello curently I was not able to figure out how to log Set and Map in apex code.
Yes there some king of workaround?

thanks

**Comment by jongpie** [comment, marked as answer]
Hi @mmakara2021 - great question, unfortunately there is not currently a built-in way to log Set and Map. [In v4.3.0](https://github.com/jongpie/NebulaLogger/releases/tag/v4.3.0), I added support for `List<SObject>`, but I still have not implemented support for Set or Map yet. I've created issue #222 to add this functionality for `Set<SObject>` and `Map<Id, SObject>` in a future release - in the meantime, you can try these workarounds:

```java
// Workaround for logging a Set<SObject>
Set<SObject> someRecordSet = new Set<SObject>();
Logger.info('logging a set by converting to a list', new List<SObject>(someRecordSet));

// Workaround for logging a Map<Id, SObject>
Map<Id, SObject> someRecordMap = new Map<Id, SObject>();
Logger.info('logging a set by converting to a list', new List<SObject>(someRecordMap?.keySet()));
````

**Reply by mmakara2021** [reply]
For the map it does not work giving me error:
Error:(56, 69) Invalid initializer type Set<Id> found for List<SObject>: expected an Integer or a List of type SObject

**Reply by jongpie** [reply]
@mmakara2021 sorry about that! I had 1 mistake in my original answer - it should use the `values()` method, instead of `keySet()` method, on the Map. Here's the updated/corrected sample:

```java
// Workaround for logging a Set<SObject>
Set<SObject> someRecordSet = new Set<SObject>();
Logger.info('logging a set by converting to a list', new List<SObject>(someRecordSet));

// Workaround for logging a Map<Id, SObject>
Map<Id, SObject> someRecordMap = new Map<Id, SObject>();
Logger.info('logging a set by converting to a list', new List<SObject>(someRecordMap?.values()));
````
